src/internal/operators/share.ts

```diff
@@ -1,14 +1,50 @@
-import { Observable } from '../Observable';
-import { multicast } from './multicast';
-import { refCount } from './refCount';
+/** @prettier */
+
 import { Subject } from '../Subject';
 
-import { MonoTypeOperatorFunction } from '../types';
+import { MonoTypeOperatorFunction, OperatorFunction, SubjectLike } from '../types';
+import { Subscription } from '../Subscription';
+import { from } from '../observable/from';
+import { operate } from '../util/lift';
 
-function shareSubjectFactory() {
-  return new Subject<any>();
+interface ShareOptions<T, R> {
```

Comment:
issue: My understanding is that typing this using `interface` and not exporting it will be a problem for consumers of the lib. Types declared using `type` are anonymous, but types declared with `interface` are not. See [this playground snippet](selects what sort of values are emitted from the result) - from [this blog post](https://ncjamieson.com/prefer-interfaces/) - that simulates the situation using an IIFE. However, I don't understand why this isn't effecting problems with the tests. 🤷‍♂️ Still a concern, though.